helper/resource/compatibility.go

```diff
@@ -0,0 +1,13 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package resource
+
+// Deprecated. This is an undocumented compatibility flag. When
+// `RefreshAfterApply` is set to non-empty, a `Config`-mode test step will
+// invoke a refresh before successful completion. This is intended as a
+// compatibility measure for test cases that have different -- but
+// semantically-equal -- state representations in their test steps. When
+// comparing two states, the testing framework is not aware of semantic
+// equality or set equality, as that would rely on provider logic.
+var RefreshAfterApply string
```

Comment:
When I started playing with this, I deliberately chose a very long variable name as a feature flag. The thought is that you'd add this as one more `ldflags` value in a `testacc` Make target:
```
-X github.com/hashicorp/terraform-plugin-testing/helper/resource.DeprecatedCompatibilityFlagRefreshAfterApply=unlocked
```

The `ldflags` bit is a weakly-held opinion. I would be receptive to using a simpler name and a `TF_*` environment variable instead.

Reply:
Personally I'd lean toward an environment variable, just being consistent with existing configuration options: `TF_ACC_REFRESH_AFTER_APPLY` or something of the like 😆